docs/rackhd/graph_progress.rst

```diff
@@ -0,0 +1,146 @@
+Workflows Progress
+---------------------------
+
+RackHD workflow progress feature provide message notification mechanism to indicate status of an active workflow or task. User can get to know what have been done and what is left for an active workflow or task with progress message. It is helpful when user is running time-consuming workflows like OS installation or secure erase.
+
+Progress Message Publishment Criteria
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+RackHD will publish a workflow progress message if any of below events happened:
+
+* Workflow started or finished events
+* Task started or finished events
+* RackHD marked important milestone events achieved for an active long-run task.
+
+  In some cases RackHD can't easily get progress information, some milestones are created to divide a task into several small sections. Progress message will be sent if any of those milestones is achieved.
+
+* Progress timer timeout for an active long-run task.
+
+  In some cases progress of a task is continuous, RackHD can't keep sending progress information all the time. Thus a timer is set and progress information is publish with a fixed interval.
```

Comment:
The wording here is a little hard for me to understand. Could you please rephrase it?